WordPress/Classes/Services/AccountService.m

```diff
@@ -287,4 +287,24 @@ - (void)updateAccount:(WPAccount *)account withUserDetails:(RemoteUser *)userDet
     [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
 }
 
+- (void)purgeSelfHostedAccount:(WPAccount *)account
+{
+    if (!account.isWpcom && [account.blogs count] == 0) {
+        [self.managedObjectContext deleteObject:account];
+    }
+}
+
+- (void)purgeJetpackAccount:(WPAccount *)account
```

Comment:
I'd consider checking we're dealing with a jetpack account inside this method, or if we consider that to be a pre-condition, to have at least an assertion to check it.